src/main/java/io/rocketpartners/cloud/action/rest/RestPostAction.java

```diff
@@ -47,18 +47,104 @@
    protected boolean strictRest     = true;
    protected boolean expandResponse = true;
 
+   protected List<String> upsert(Request req, Collection collection, ArrayNode nodes) throws Exception
+   {
+      Map<String, Object> mapped;
+      Set copied = new HashSet();
+      List<Map<String, Object>> maps = new ArrayList<Map<String, Object>>();
+      for (ObjectNode node : (List<ObjectNode>) ((ArrayNode) nodes).asList())
+      {
+         mapped = new HashMap();
+         for (Attribute attr : collection.getEntity().getAttributes())
```

Comment:
Could you please explain what the following lines of code will do? So that I can extract these into a common method to avoid duplication

Reply:
So inversion is split between "Collections" and "Tables".  Collections represent the data as delivered to the API client in JSON form.  Tables represent the actual db structure.  Collections can have different attribute names than their corresponding table columns etc.  This RestPostAction.upsert() effectively translates the json attribute names and types into their actual db names and types.  Everything that is submitted to one of the Db.upsert/delete/select methods has been translated in such a manner.  These Rest*Action methods are supposed to make it easy to plug a new Db subclass implementation in and expose the backend resource with minimal effort.